tests/Agent/IntegrationTests/UnboundedIntegrationTests/Elasticsearch/ElasticsearchAsyncTests.cs

```diff
@@ -0,0 +1,280 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+
+using System.Collections.Generic;
+using System.Linq;
+using MultiFunctionApplicationHelpers;
+using NewRelic.Agent.IntegrationTestHelpers;
+using NewRelic.Agent.UnboundedIntegrationTests.ElasticsearchTests;
+using NewRelic.Testing.Assertions;
+using StackExchange.Redis;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NewRelic.Agent.UnboundedIntegrationTests.Elasticsearch
+{
+    public abstract class ElasticsearchAsyncTestsBase<TFixture> : ElasticsearchTestsBase<TFixture>
+        where TFixture : ConsoleDynamicMethodFixture
+    {
+        const string SYNC_MODE = "async";
+
+        protected ElasticsearchAsyncTestsBase(TFixture fixture, ITestOutputHelper output, string clientType) : base(fixture, output, clientType, SYNC_MODE)
```

Comment:
If we have separate classes for testing async and sync, we will spin up a separate process for each one. Is it possible to test the sync calls and async calls during the same run of the application? If we can do that, it might help with this test suite running faster.

Reply:
True. I was trying to maximize code reuse. I could still do that with multiple tests with a little refactoring. Surprisingly, the whole thing completes in 30 seconds on my machine.

Reply:
Yeah, I think the impact is more noticeable in our CI because things are not run in parallel there.

Reply:
I'm also fine with merging the tests as is, and then with a separate PR we could see how much of an impact combining the sync/async tests have on the CI testing time.

Reply:
I should be able to get it done today. If not we can split it out.

Reply:
If you do get it done it will be interesting to know just how much of an impact it has on the timing of our integration tests (even if it's just this one sample point).

Reply:
Now that I've thought about it, I'm not sure how to accomplish this with the ConsoleMF pattern. We need a different set of library calls for sync and async, but we set all the commands to run at process creation time. We can brainstorm in pairing.